File: cpp/2873.cpp
#define max(a, b) ((a > b) ? a : b)

class Solution {
public:
    long long maximumTripletValue(vector<int>& nums) {
        long long res = 0, diff = 0;
        int i = 0;
        for (int j : nums) {
            res = max(res, diff * j);
            diff = max(diff, i - j);
            if (j > i)
                i = j;
        }
        return res;
    }
};

File: python3/2873.py
class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        res = 0
        diff = 0
        i = 0
        for j in nums:
            res = max(res, diff * j)
            diff = max(diff, i - j)
            i = max(i, j)
        return res
